src/main/java/jeff/task/TaskList.java

```diff
@@ -0,0 +1,195 @@
+package jeff.task;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import jeff.exception.FileCorruptException;
+import jeff.exception.JeffException;
+
+/**
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    private final ArrayList<Task> tasks = new ArrayList<>();
+
+    public TaskList() {}
+
+    /**
+     * Constructor for the TaskList Class.
+     * Takes in a scanner, categorises the strings into their task types and adds the tasks to the task list.
+     *
+     * @param scanner Scanner with task strings.
+     * @throws JeffException if the file is corrupt.
+     */
+    public TaskList(Scanner scanner) throws JeffException {
+        try {
+            // Read every line in the scanner
+            while (scanner.hasNext()) {
+                // Get the task information
+                String[] taskParts = scanner.nextLine().split(" \\| ");
+
+                // Check if the data file is corrupted
+                if (taskParts.length < 3) {
+                    throw new FileCorruptException();
+                }
```

Comment:
I like the input validation sprinkled throughout the project to defend against the myriad of possible bad inputs. Nice job!